library/alloc/src/collections/linked_list.rs

```diff
@@ -392,17 +404,17 @@ impl<T> LinkedList<T> {
 
             second_part
         } else {
-            mem::replace(self, LinkedList::new())
+            mem::replace(self, LinkedList::new_in(self.alloc.clone()))
         }
     }
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
-impl<T> Default for LinkedList<T> {
+impl<T, A: Allocator + Clone + Default> Default for LinkedList<T, A> {
```

Comment:
We usually just implement default for `Global` instead of all allocators, because that would cause inference failures in many cases.

Reply:
Concretely, every program using `LinkedList::default()` that doesn't explicitly assign the list to `LinkedList<_>` (for example by passing it to a function or storing it in a field) will not infer an allocator, as type inference does not take into account default type parameters.